File: orders/src/main/java/com/ecommerce/orders/controller/CartController.java
package com.ecommerce.orders.controller;

import com.ecommerce.orders.entity.Cart;
import com.ecommerce.orders.repo.CartRepository;

import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.*;

@RestController
@RequestMapping("/cart")
public class CartController {

	@Autowired
    private CartRepository cartRepository;

    @GetMapping
    public ResponseEntity<?> getCart() {
        return ResponseEntity.ok(cartRepository.findAll());
    }

    @PostMapping("/add")
    public ResponseEntity<Cart> addCart(@RequestBody Cart cart) {
        Cart savedCart = cartRepository.save(cart);
        return ResponseEntity.ok(savedCart);
    }
}

File: order/order/src/main/kotlin/com/ecommerce/order/controllers/CartController.kt
package com.ecommerce.order.controllers

import com.ecommerce.order.entity.Cart
import com.ecommerce.order.repo.CartRepository
import org.springframework.http.ResponseEntity
import org.springframework.web.bind.annotation.*

@RestController
@RequestMapping("/cart")
class CartController (
    private val cartRepository: CartRepository
){

    @GetMapping("/")
    fun getCart(): ResponseEntity<Any?> {
        return ResponseEntity.ok(cartRepository.findAll())
    }

    @PostMapping("/add")
    fun addCart(@RequestBody cart: Cart): ResponseEntity<Cart> {
        val savedCart = cartRepository.save(cart)
        return ResponseEntity.ok(savedCart)
    }

}